balancer/weightedroundrobin/balancer_test.go

```diff
@@ -129,98 +157,44 @@ func svcConfig(t *testing.T, wrrCfg iwrr.LBConfig) string {
-		EnableOOBLoadReport:     boolp(false),
-		OOBReportingPeriod:      durationp(5 * time.Millisecond),
-		BlackoutPeriod:          durationp(0),
-		WeightExpirationPeriod:  durationp(time.Minute),
-		WeightUpdatePeriod:      durationp(weightUpdatePeriod),
-		ErrorUtilizationPenalty: float64p(0),
-	}
-	oobConfig = iwrr.LBConfig{
-		EnableOOBLoadReport:     boolp(true),
-		OOBReportingPeriod:      durationp(5 * time.Millisecond),
-		BlackoutPeriod:          durationp(0),
-		WeightExpirationPeriod:  durationp(time.Minute),
-		WeightUpdatePeriod:      durationp(weightUpdatePeriod),
-		ErrorUtilizationPenalty: float64p(0),
-	}
-)
-
-func (s) TestBalancer_OneAddress_ReportingDisabled(t *testing.T) {
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
-
-	srv := startServer(t, reportNone)
-
-	sc := svcConfig(t, perCallConfig)
-	if err := srv.StartClient(grpc.WithDefaultServiceConfig(sc)); err != nil {
-		t.Fatalf("Error starting client: %v", err)
-	}
-
-	// Perform many RPCs to ensure the LB policy works with 1 address.
-	for i := 0; i < 100; i++ {
-		if _, err := srv.Client.EmptyCall(ctx, &testpb.Empty{}); err != nil {
-			t.Fatalf("Error from EmptyCall: %v", err)
-		}
-		time.Sleep(time.Millisecond) // Delay; test will run 100ms and should perform ~10 weight updates
-	}
-}
-
-func (s) TestBalancer_OneAddress_ReportingEnabledPerCall(t *testing.T) {
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
-
-	srv := startServer(t, reportCall)
-
-	sc := svcConfig(t, perCallConfig)
-	if err := srv.StartClient(grpc.WithDefaultServiceConfig(sc)); err != nil {
-		t.Fatalf("Error starting client: %v", err)
-	}
-
-	// Perform many RPCs to ensure the LB policy works with 1 address.
-	for i := 0; i < 100; i++ {
-		srv.callMetrics.SetQPS(float64(i))
-		if _, err := srv.Client.EmptyCall(ctx, &testpb.Empty{}); err != nil {
-			t.Fatalf("Error from EmptyCall: %v", err)
-		}
-		time.Sleep(time.Millisecond) // Delay; test will run 100ms and should perform ~10 weight updates
+// Tests basic functionality with one address.  With only one address, load
+// reporting doesn't affect routing at all.
+func (s) TestBalancer_OneAddress(t *testing.T) {
+	testCases := []struct {
+		rt reportType
+	}{
+		{rt: reportNone},
+		{rt: reportCall},
+		{rt: reportOOB},
 	}
-}
 
-func (s) TestBalancer_OneAddress_ReportingEnabledOOB(t *testing.T) {
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("reportType:%v", tc.rt), func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+			defer cancel()
 
-	srv := startServer(t, reportOOB)
+			srv := startServer(t, tc.rt)
 
-	sc := svcConfig(t, oobConfig)
-	if err := srv.StartClient(grpc.WithDefaultServiceConfig(sc)); err != nil {
-		t.Fatalf("Error starting client: %v", err)
-	}
+			sc := svcConfig(t, perCallConfig)
```

Comment:
I think this config needs to be a knob as well (corresponding to the reporting type knob).

Reply:
Sorry, not following this?

Reply:
Ahhhh, good call; done.